iree_tests/compile_tests.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2024 Advanced Micro Devices
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from pathlib import Path
+import subprocess
+import sys
+
+THIS_DIR = Path(__file__).parent
+REPO_ROOT = THIS_DIR.parent
+
+# Test directory generated by iree_tests/onnx/scripts/generate_onnx_tests.py
+# TODO(scotttodd): flag to run across any compatible test suite folder
+TEST_SOURCE_ROOT = REPO_ROOT / "iree_tests/onnx/node/generated"
+
+# Write lists of tests that passed/failed to compile.
+COMPILE_SUCCESSES_FILE = REPO_ROOT / "iree_tests/onnx/node/compile_successes.txt"
+COMPILE_FAILURES_FILE = REPO_ROOT / "iree_tests/onnx/node/compile_failures.txt"
```

Comment:
This will turn into some sort of configuration file in the test runner, starting in this repo and then moving into downstream projects. IREE can run the tests it wants with its own out-of-band labels for which test cases pass/fail, iree-amd-aie can do the same, etc.